Contest_API.md

```diff
@@ -116,27 +116,36 @@ their expected behavior, if implemented.
     It can be used to request a whole collection or a specific element.
   - `POST`
     Create a new element. This can only be called on a collection
-    endpoint. No `id` attribute should be specified as it is up to the
-    server to assign one, which is returned in the location header.
+    endpoint. When using most [roles](#Roles) the `id` attribute may not
+    be specified as it is up to the server to assign one. When using an
+    `admin` role you may specify a new, unused id.
+    The response will contain a `Location` header pointing to the newly
+    created element and the response body will contain the initial
+    state of the element.
   - `PUT`
     Replaces a specific element. This method is idempotent and can only
     be called on a specific element and replaces its contents with the
     data provided. The payload data must be complete, i.e. no partial
     updates are allowed. The `id` attribute cannot be changed: it does
     not need to be specified (other than in the URL) and if specified
     different from in the URL, a `409 Conflict` HTTP code should be
```

Comment:
Why do we want to allow not specifying the `id`? Is there any use case where this would be beneficial? If not. I would prefer to require it. (That is kinda what the HTTP specs says after all)

Reply:
As said above: I'm fine with requiring the ID but I am not fine with allowing it to be different than what it was before.

Reply:
I don't know why this was done originally, but I agree it should be required in PUT, and I agree with Nicky that it must match the URI.